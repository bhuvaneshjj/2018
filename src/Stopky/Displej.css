.Displej, .Displej-segment {
	font-family: monospace;
	font-weight: bold;
}

.Displej-segment {
	background: black;
	color: white;
	padding: 1ex;
	border-radius: 10px;
}


