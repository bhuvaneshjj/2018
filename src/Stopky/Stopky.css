.Stopky {
  text-align: center;
}

.Stopky-logo {
  animation: Stopky-logo-spin infinite 20s linear;
  height: 80px;
}

.Stopky-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.Stopky-title {
  font-size: 1.5em;
}

.Stopky-intro {
  font-size: large;
}

@keyframes Stopky-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
