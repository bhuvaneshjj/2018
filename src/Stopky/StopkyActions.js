export const stopkyStart = base => {
  return {
    type: 'STOPKY_START',
    base: base
  };
};

export const stopkyStop = () => {
  return {
    type: 'STOPKY_STOP'
  };
};
