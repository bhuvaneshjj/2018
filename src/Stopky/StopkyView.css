.StopkyView {
  font-size: larger;
}

.StopkyView-mezera {
	padding-bottom: 20px;
}