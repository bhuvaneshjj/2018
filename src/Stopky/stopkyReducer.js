const initialState = {
	'running': false,
	'base': null
};

const stopkyReducer = (state = initialState, action) => {
	switch (action.type) {
	case 'STOPKY_START':
		return {
			...state,
			'running': true,
			'base': new Date()
		};
	case 'STOPKY_STOP':
		return {
			...state,
			'running': false
		};
	default:
	    return state;
	}
}

export default stopkyReducer;