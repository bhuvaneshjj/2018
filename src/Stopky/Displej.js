import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import './Displej.css';

const Displej = ({ duration }) => {
  let hours, mins, secs, subsecs;
  if (duration === null) {
    hours = '-';
    mins = '--';
    secs = '--';
    subsecs = '--';
  } else {
    hours = duration.hours().toString();
    mins = duration.minutes().toString();
    secs = duration.seconds().toString();
    subsecs = duration.milliseconds().toString();
    if (subsecs.length < 2) {
      subsecs = subsecs + '0';
    } else {
      subsecs = subsecs.slice(0, 2);
    }
  }

  return (
    <span className="Displej">
      <span className="Displej-segment">{hours}</span>
      {':'}
      <span className="Displej-segment">{mins}</span>
      {':'}
      <span className="Displej-segment">{secs}</span>
      {','}
      <span className="Displej-segment">{subsecs}</span>
    </span>
  );
};

Displej.propTypes = {
  duration: PropTypes.object
};

export default Displej;
