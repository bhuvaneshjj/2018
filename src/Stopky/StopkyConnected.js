import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Stopky from './Stopky';
import { stopkyStart, stopkyStop } from './StopkyActions';

const mapStateToProps = state => ({
  base: new Date(state.stopky.base),
  running: state.stopky.running,
  startEnabled: state.stopky.running === false,
  mezicasEnabled: state.stopky.running === true,
  stopEnabled: state.stopky.running === true
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  onStart: base => {
    dispatch(stopkyStart(base));
  },
  onStop: () => {
    dispatch(stopkyStop());
  },
  onAddMezicas: ownProps.onAddMezicas
});

const StopkyConnected = connect(mapStateToProps, mapDispatchToProps)(Stopky);

StopkyConnected.propTypes = {
  onAddMezicas: PropTypes.func.isRequired
};

export default StopkyConnected;
