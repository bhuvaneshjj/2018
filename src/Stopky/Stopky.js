import { connect } from 'react-redux';
import StopkyView from './StopkyView';
import { stopkyStart, stopkyStop } from './StopkyActions';

const mapStateToProps = state => {
  return { base: state.base };
};

const mapDispatchToProps = dispatch => {
  return {
    startAction: base => {
      dispatch(stopkyStart(base));
    },
    stopAction: () => {
      dispatch(stopkyStop());
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(StopkyView);
