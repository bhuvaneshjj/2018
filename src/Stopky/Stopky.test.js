import React from 'react';
import ReactDOM from 'react-dom';
import Stopky from './Stopky';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Stopky />, div);
});