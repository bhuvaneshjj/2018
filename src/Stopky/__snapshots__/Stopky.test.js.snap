// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`po startu 1`] = `
<div
  className="Stopky"
>
  <div
    className="Stopky-mezera"
  >
    <span
      className="Displej"
    >
      <span
        className="Displej-segment"
      >
        0
      </span>
      :
      <span
        className="Displej-segment"
      >
        00
      </span>
      :
      <span
        className="Displej-segment"
      >
        00
      </span>
      ,
      <span
        className="Displej-segment"
      >
        00
      </span>
    </span>
  </div>
  <div>
    <div
      className="btn-toolbar"
      role="toolbar"
    >
      <button
        className="btn btn-success"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Start
      </button>
      <button
        className="btn btn-info"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Mezičas
      </button>
      <button
        className="btn btn-danger"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Stop
      </button>
    </div>
  </div>
</div>
`;

exports[`po stopce 1`] = `
<div
  className="Stopky"
>
  <div
    className="Stopky-mezera"
  >
    <span
      className="Displej"
    >
      <span
        className="Displej-segment"
      >
        0
      </span>
      :
      <span
        className="Displej-segment"
      >
        00
      </span>
      :
      <span
        className="Displej-segment"
      >
        00
      </span>
      ,
      <span
        className="Displej-segment"
      >
        00
      </span>
    </span>
  </div>
  <div>
    <div
      className="btn-toolbar"
      role="toolbar"
    >
      <button
        className="btn btn-success"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Start
      </button>
      <button
        className="btn btn-info"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Mezičas
      </button>
      <button
        className="btn btn-danger"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Stop
      </button>
    </div>
  </div>
</div>
`;

exports[`před startem 1`] = `
<div
  className="Stopky"
>
  <div
    className="Stopky-mezera"
  >
    <span
      className="Displej"
    >
      <span
        className="Displej-segment"
      >
        -
      </span>
      :
      <span
        className="Displej-segment"
      >
        --
      </span>
      :
      <span
        className="Displej-segment"
      >
        --
      </span>
      ,
      <span
        className="Displej-segment"
      >
        --
      </span>
    </span>
  </div>
  <div>
    <div
      className="btn-toolbar"
      role="toolbar"
    >
      <button
        className="btn btn-success"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Start
      </button>
      <button
        className="btn btn-info"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Mezičas
      </button>
      <button
        className="btn btn-danger"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Stop
      </button>
    </div>
  </div>
</div>
`;
